proposals/0023-asynchronous-program-execution-and-broadcast.md

```diff
@@ -0,0 +1,148 @@
+
+## New Terminology
+
+* UserBlock - a block full of none vote transactions.
+
+* UserBlockEntry - a hash of the block that was proposed
+
+* UserBlockSlot - a slot measuring half a normal slot in ticks.
+
+* Leader - the current leader for the slot that will propose a PoH
+ledger full of Votes and UserBlockEntry
+
+* Builder - a node that is scheduled to propose a block with non
+vote transactions
+
+## Detailed Design
+
+### Overview
+Leaders are scheduled to propose blocks as they are currently by
+the LeaderSchedule.
+
+Builder's are scheduled along side leaders to propose blocks. N
+number of builders can be scheudled per slot.  Builder are
+scheudled at 2x the rate of leaders.  So for every leader slot,
+there are N builders producing 2 UserBlocks.
+
+While a leader is scheduled, they receive and encode votes as normal.
+Any well formed UserBlocks that were received from the previous
+UserBlockSlot are added to the leaders PoH ledger as UserBlockEntry.
+
+The N concurrent Builder have 200ms slots to create blocks out
+of user transactions. These are transmitted to the network.  The
+leader receives and decodes them and generates a UserBlockEntry,
+and adds it to PoH as soon as the leaders PoH has passed the
+UserBlockSlot.
+
+### Fork Choice
+
+If a validator doesn't have the builder's UserBlock, the validator
+doesn't vote on the proposed block and tries to repair the UserBlock.
+That fork is ignored for fork choice.
+
+Otherwise the validator evaluates the forks as usual. 
+
+### UserBlock execution
+
+Each UserBlock is assumed to have been created simultaneously
+during its scheduled slot.  For each UserBlock, the transactions
+are ordered by priority fee before execution. If two transactions
+from two different blocks have the same priority, the earliest
+transaction by UserBlock PoH is executed first, otherwise by which
+UserBlock appears first in the leaders PoH.
+
+### BankHash
+
+Votes must include the most recent computed BankHash along with the
+slot for the BankHash computation.  This slot is different
+from the tower vote slot.
+
+Each validator that is computing a BankHash will be able to verify
+the BankHash values created by block producers for previous slots
+because each BankHash has a link to the previous one.
+
+During verification, if a validator sees that > 1/3 of the network
+has incorrectly computed a state transition, they should halt
+immediately.
+
+### Client Confirmations
+
+Client can receive a confirmation as soon as the UserBlockEntry has
+been optimistically confirmed. Client can also wait for a status
+code weighted by stake weight, as each validator's execution catches
+up to the slot and includes a BankHash confirming the state of the
+execution.
+
+It may be sufficient to wait for 1/3+ validators executing the state
+transition instead of the full 2/3+, because if 1/3+ are incorrect
+the network will halt anyways.
+
+## Impact
```

Comment:
looks like we might want to include a bundle primitive in here too to avoid splitting up bundles that get reordered/not guaranteed to execute atomically. can imagine bundle like:
[user_high_priority_tx, arb_low_priority_tx]

Reply:
Added the BundleTransaction